js/src/sdk/models/connectedAccounts.ts

```diff
@@ -70,16 +79,31 @@ export class ConnectedAccounts {
 
     async initiate(payload: InitiateConnectionDataReq): Promise<ConnectionRequest> {
         try {
-            const {integrationId, entityId = 'default', labels,data={}, redirectUri} = payload;
-      
-            const res = await client.connections.initiateConnection({ body: {
-                integrationId,
-                entityId,
-                labels,
-                redirectUri,
-                data,
-            }  }).then(res => res.data);
-            
+            let { integrationId, entityId = 'default', labels, data = {}, redirectUri, authMode = '', authConfig = {}, appName } = payload;
+
+            if (!integrationId && authMode) {
```

Comment:
Consider extracting the integration creation logic into a separate private method for better code organization and reusability:
```typescript
private async createNewIntegration(
    appName: string,
    authMode: string,
    authConfig: AuthConfig
): Promise<string> {
    const timestamp = new Date().toISOString().replace(/[-:.]/g, "");
    const app = await this.apps.get({ appKey: appName });
    if (!app?.appId) {
        throw new Error(`Invalid app: ${appName}`);
    }
    
    const integration = await this.integrations.create({
        appId: app.appId,
        name: `integration_${timestamp}`,
        authScheme: authMode,
        authConfig,
        useComposioAuth: false,
    });
    
    return integration?.id ?? throw new Error('Failed to create integration');
}
```

Reply:
Not required.